src/methods/aggregate.jl

```diff
@@ -259,10 +264,10 @@ Disaggregate array `src` to array `dst` by some scale, using `method`.
 Note: currently it is faster to aggregate over memory-backed arrays.
 Use [`read`](@ref) on `src` before use where required.
 """
-function disaggregate!(locus::Locus, dst::AbstractRaster, src, scale)
+function disaggregate!(dst::AbstractRaster, src, scale)
     disaggregate!((locus,), dst, src, scale)
 end
-function disaggregate!(loci::Tuple{Locus,Vararg}, dst::AbstractRaster, src, scale)
+function disaggregate!(dst::AbstractRaster, src, scale)
     intscale = _scale2int(DisAg(), dims(src), scale)
     broadcast!(dst, CartesianIndices(dst)) do I
```

Comment:
It must be faster to loop through the `src` instead like so since we know it is smaller.

By looping through `dst` we have lots of unnecessary calls to `upsample` and lookups in `src`.

```suggestion
    for I in CartesianIndices(src)
        upper = upsample.(Tuple(I), intscale)
        lower = upper .+ intscale .- 1
        I_dst = map(:, upper, lower)
        val = src[I]
        val === missingval(src) ? missingval(dst) : val
        view(dst, I_dst...) .= val
    end
    return dst
```

Reply:
Can't comment on unchanged lines, but if we implement this then `downsample` isn't used anywhere anymore.

Reply:
Right yes it will be faster. I think the hidden reason again is that wont work on a GPU or a disk array and I was trying to be generic

Reply:
We could use Flatten.jl so see if there is an array inside whatever wrappers, and if it is do the loop, if not the broadcast

Reply:
We need DiskArraysKernelAbstractions then we could just do all of this with KernelAbstractions, and use Stencils.jl too to make it even faster

Reply:
What about this woulnd't work with on GPU? We're just viewing into an array and filling it with some value, right?

Reply:
Its all scalar indexing. We need to launch single kernel to do the whole lot in one go.

Reply:
broacasts handle kernel launches for us. If you want to write manual code like that with blocks with need KernelAbstractionsjl

Reply:
Another idea is to do some clever reshapes and permutes to be able to broadcast without having to do indexing math ourselves. I tried this implementation and it works, but it's actually a little slower than the original. I guess the reshaping makes indexing that much slower.

 ```julia
   intscale = _scale2int(DisAg(), dims(src), scale)
    n = length(intscale)
    reshape_dims = ntuple(i -> mod(i, 2) == 0 ? size(src)[(i ÷ 2)] : intscale[(i ÷ 2) + 1], Val(n*2))
    permute_order = (range(2; step = 2, length = n)..., range(1, step = 2, length = n)...)
    dst_reshaped = Base.PermutedDimsArray(Base.reshape(dst, reshape_dims), permute_order)
    broadcast!(dst_reshaped, parent(src)) do val
        val === missingval(src) ? missingval(dst) : val
    end
    return dst
```